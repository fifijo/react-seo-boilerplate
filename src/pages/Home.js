import React, { Fragment } from 'react'

const home = () => {
  return(
    <Fragment>
      <h1>Home page</h1>
    </Fragment>
  )
}
  
export default home