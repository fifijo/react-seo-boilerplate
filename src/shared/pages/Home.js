import React, { PureComponent , Fragment } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Helmet from 'react-helmet'
import { fetchData } from 'shared/state/actions/circuits'
import withLoading from 'shared/hoc/withLoading'

class Home extends PureComponent {
    componentDidMount( ) {
        (this.props.data.length <= 0) ? this.props.fetchData() : false
    }

    render() {
        const { data } = this.props

        return (
            <Fragment>
                <Helmet>
                    <title>Home Page</title>
                </Helmet>
                <ul>
                    { data.map( ( { circuitId, circuitName, Location } ) => (
                        <li key={ circuitId } >{ circuitName } - { Location.locality }, { Location.country }</li>
                    ) ) }
                </ul>
            </Fragment>
        );
    }
}

Home.displayName = 'Home'

Home.propTypes = {
   fetchData: PropTypes.func.isRequired,
   data: PropTypes.array.isRequired,
   isLoading: PropTypes.bool.isRequired,
   hasError: PropTypes.bool.isRequired
}

Home.serverFetch = fetchData

const mapStateToProps = (state) => ({
   data: state.circuits.data,
   isLoading: state.circuits.isLoading,
   hasError: state.circuits.hasError
})

const mapDispatchToProps = {
    fetchData
}

export default connect(mapStateToProps, mapDispatchToProps)(withLoading(Home))
