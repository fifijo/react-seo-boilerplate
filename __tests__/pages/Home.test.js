import React from 'react'
import { mount, render } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import Home from 'shared/pages/Home'

import createStore from 'shared/state'

const store = createStore(window.REDUX_DATA)

describe('Home page', () => {
  it('should mount correctly', () => {
    expect(
      mount(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>
      )
    ).toMatchSnapshot()
  })

  it('should render correctly', () => {
    expect(
      render(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>
      )
    ).toMatchSnapshot()
  })
})
