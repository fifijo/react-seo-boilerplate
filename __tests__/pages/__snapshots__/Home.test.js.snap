// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should mount correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Home)>
        <Home
          data={Array []}
          fetchData={[Function]}
          hasError={false}
          isLoading={true}
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
              title="Home Page"
              titleAttributes={Object {}}
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
                title="Home Page"
                titleAttributes={Object {}}
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <h2>
            This is the home page filled with fake data
          </h2>
          <ul />
        </Home>
      </Connect(Home)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Home page should render correctly 1`] = `
Array [
  <h2>
    This is the home page filled with fake data
  </h2>,
  <ul />,
]
`;
